File: optimus-fe/src/types/observability.ts
export interface Page<T>{items:T[];total:number;page:number;page_size:number}
export interface NamedRef{id:number;name:string}
export interface DatasourceSummary{id:number;name:string;base_url:string;auth_type:'none'|'basic'|'bearer';http_credential?:NamedRef;cluster?:NamedRef;tls_skip_verify:boolean;has_custom_ca:boolean;description:string;created_at:string;updated_at:string}
export type DatasourceDetail=DatasourceSummary
export interface SaveDatasource{name:string;base_url:string;auth_type:'none'|'basic'|'bearer';http_credential_id?:number;cluster_id?:number;tls_skip_verify:boolean;custom_ca_pem?:string;description:string;clear_http_credential?:boolean;clear_cluster?:boolean;clear_custom_ca?:boolean}
export interface Query{ref_id:string;promql:string}
export interface InstantBatch{datasource_id:number;time?:string;enrich_assets:boolean;queries:Query[]}
export interface RangeBatch{datasource_id:number;start:string;end:string;step:string;enrich_assets:boolean;queries:Query[]}
export interface Sample{timestamp:number;value:string}
export interface Series{labels:Record<string,string>;samples:Sample[]}
export interface NormalizedResult{result_type:'vector'|'matrix'|'scalar'|'string';series:Series[];scalar?:Sample;string?:Sample;warnings?:string[]}
export interface ItemError{code:number;message:string;message_key?:string}
export interface QueryItemResult{ref_id:string;result?:NormalizedResult;error?:ItemError}
export interface BatchResult{results:QueryItemResult[]}
export interface PanelInput{datasource_id:number;title:string;panel_type:'time_series'|'stat'|'table';promql:string;unit:string;legend:string;sort_order:number;width:6|12}
export interface Panel extends PanelInput{id:number;dashboard_id:number;created_at:string;updated_at:string}
export interface SaveDashboard{name:string;description:string;refresh_interval_s:number;time_range:'15m'|'1h'|'6h'|'24h'|'7d';panels:PanelInput[]}
export interface Dashboard extends Omit<SaveDashboard,'panels'>{id:number;created_by_user_id?:number;created_at:string;updated_at:string;panels:Panel[]}
export interface BuiltinVariable{name:string;label:string;required:boolean}
export interface BuiltinPanel{ref_id:string;title_key:string;panel_type:'time_series'|'stat'|'table';promql:string;unit:string;sort_order:number;width:6|12}
export interface BuiltinDashboard{code:string;title_key:string;variables:BuiltinVariable[];panels:BuiltinPanel[]}

File: optimus-be/internal/models/observability.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type ObservabilityDatasource struct {
	ID                                   uint64 `gorm:"primaryKey"`
	Name, BaseURL, AuthType, Description string
	HTTPCredentialID, ClusterID          *uint64
	TLSSkipVerify                        bool
	CustomCAPEM                          *string `gorm:"column:custom_ca_pem"`
	CreatedByUserID                      *uint64
	CreatedAt, UpdatedAt                 time.Time
	DeletedAt                            gorm.DeletedAt `gorm:"index"`
}

func (ObservabilityDatasource) TableName() string { return "observability_datasources" }

type ObservabilityDashboard struct {
	ID                           uint64 `gorm:"primaryKey"`
	Name, Description, TimeRange string
	RefreshIntervalS             int
	CreatedByUserID              *uint64
	CreatedAt, UpdatedAt         time.Time
	DeletedAt                    gorm.DeletedAt `gorm:"index"`
}

func (ObservabilityDashboard) TableName() string { return "observability_dashboards" }

type ObservabilityPanel struct {
	ID, DashboardID, DatasourceID uint64
	Title, PanelType              string
	PromQL                        string `gorm:"column:promql"`
	Unit, Legend                  string
	SortOrder, Width              int
	CreatedAt, UpdatedAt          time.Time
}

func (ObservabilityPanel) TableName() string { return "observability_panels" }
